frontend/device/kindle/powerd.lua

```diff
@@ -43,7 +53,7 @@ function KindlePowerD:init()
             -- "Discharging" when discharging
             -- "Full" when full
             -- "Charging" when charging via DCP
-            return this:read_str_file(this.aux_batt_status_file) ~= "Discharging"
+            return this:read_str_file(this.aux_batt_status_file) == "Charging"
```

Comment:
Is this change quite right? Not discharging feels more accurate given that full is a type of charging.

Reply:
As shortly mentioned in the commit message, there are more states.

There is also

```
[root@kindle root]# cat /sys/devices/platform/soda/power_supply/soda_fg/status
Not charging
```

Which means basically that the aux battery is idle. It does not get charged, it is not full (which I also would not consider as charging since it can not longer accept more charges, but no hard feelings regarding that point), and it does not charge the main battery.

And it flickers between 'Discharging' and 'Not charging' all the time.

Reply:
Please add that to the comment.

Reply:
That said, given the actual behavior you just described, your approach sounds more sensible for this specific use-case ;).